tools/deleteorphanlogcommand.class.php

```diff
@@ -0,0 +1,193 @@
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+
+use Glpi\Console\AbstractCommand;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Helper\ProgressBar;
+
+
+
+class DeleteOrphanLogCommand extends AbstractCommand {
+
+   protected $requires_db_up_to_date = false;
+
+   protected function configure() {
+      parent::configure();
+
+      $this->setName('glpi:tools:delete_orphan_logs');
+      $this->setAliases(
+         [
+            'tools:delete_orphan_logs',
+         ]
+      );
+      $this->setDescription(__('Delete orphan log'));
+
+      $this->addOption(
+         'dry-run', '',
+         InputOption::VALUE_NONE,
+         __('Only print the results whitout effective deletion')
+      );
+
+   }
+
+   protected function execute(InputInterface $input, OutputInterface $output) {
+
+      $no_interaction = $input->getOption('no-interaction');
+      if (!$no_interaction) {
+         // Ask for confirmation (unless --no-interaction)
+         $output->writeln([
+               __('You are about to delete orphan logs of GLPI log table (glpi_logs).'),
+               __('The first step check if orphans exists in database.'),
+               __('The final step is the definitive deletion of orphans (unless if --dry-run arguments is used used).')
+            ]
+         );
+
+         /**
+          * @var QuestionHelper $question_helper
+          */
+         $question_helper = $this->getHelper('question');
+         $run = $question_helper->ask(
+            $input,
+            $output,
+            new ConfirmationQuestion(
+               '<comment>' . __('Do you want to launch operation ?') . ' [yes/No]</comment>',
+               false
+            )
+         );
+         if (!$run) {
+            $output->writeln(
+               '<comment>' . __('Operation aborted.') . '</comment>',
+               OutputInterface::VERBOSITY_VERBOSE
+            );
+            return 0;
+         }
+      }
+
+      return $this->deleteOrphan($input, $output);
+   }
+
+   private function deleteOrphan(InputInterface $input, OutputInterface $output): bool {
```

Comment:
As this method always return true, maybe returned value can be removed.
```suggestion
   private function deleteOrphan(InputInterface $input, OutputInterface $output) {
```

Reply:
done